src/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -0,0 +1,198 @@
+  </data>
+  <data name="ExpectedDigitNotFoundEndOfData" xml:space="preserve">
+    <value>Invalid number. Last character read: '{0}'. Expected a digit ('0'-'9') but reached end of data.</value>
+  </data>
+  <data name="ExpectedEndAfterSingleJson" xml:space="preserve">
+    <value>Expected end of json after a single value but additional data found. Last character read: '{0}'.</value>
+  </data>
+  <data name="ExpectedEndOfDigitNotFound" xml:space="preserve">
+    <value>Invalid end of number. Last character read: '{0}'. Expected a delimiter.</value>
+  </data>
+  <data name="ExpectedFalse" xml:space="preserve">
+    <value>Expected a 'false' value, but instead got '{0}'.</value>
+  </data>
+  <data name="ExpectedNextDigitComponentNotFound" xml:space="preserve">
+    <value>Invalid number. Last character read: '{0}'. Expected '.' or 'E' or 'e'.</value>
+  </data>
+  <data name="ExpectedNextDigitEValueNotFound" xml:space="preserve">
+    <value>Invalid number. Last character read: '{0}'. Expected 'E' or 'e'.</value>
+  </data>
+  <data name="ExpectedNull" xml:space="preserve">
+    <value>Expected a 'null' value, but instead got '{0}'.</value>
+  </data>
+  <data name="ExpectedSeparaterAfterPropertyNameNotFound" xml:space="preserve">
+    <value>Expected a ':' following the property name, but instead got '{0}'.</value>
+  </data>
+  <data name="ExpectedStartOfPropertyNotFound" xml:space="preserve">
+    <value>Expected '"' for start of property name. Instead reached '{0}'.</value>
+  </data>
+  <data name="ExpectedStartOfPropertyOrValueNotFound" xml:space="preserve">
+    <value>Expected a start of a property name or value after ',', but reached end of data instead.</value>
+  </data>
+  <data name="ExpectedStartOfValueNotFound" xml:space="preserve">
+    <value>Expected start of a value, but instead got '{0}'.</value>
+  </data>
+  <data name="ExpectedTrue" xml:space="preserve">
+    <value>Expected a 'true' value, but instead got '{0}'.</value>
+  </data>
+  <data name="ExpectedValueAfterPropertyNameNotFound" xml:space="preserve">
+    <value>Expected a value following the property, but instead reached end of data.</value>
+  </data>
+  <data name="FoundInvalidCharacter" xml:space="preserve">
+    <value>Expected either ',', '}}', or ']', but instead got '{0}'.</value>
+  </data>
+  <data name="InvalidCharacterWithinString" xml:space="preserve">
+    <value>Found an invalid character within a json string: '{0}'. It is expected that the string is properly escaped.</value>
+  </data>
+  <data name="InvalidEndOfJson" xml:space="preserve">
+    <value>Expected valid end of json payload with TokenType either EndArray or EndObject. Current token type is '{0}'.</value>
+  </data>
+  <data name="MaxDepthMustBePositive" xml:space="preserve">
+    <value>Max depth must be positive.</value>
+  </data>
+  <data name="ObjectDepthTooLarge" xml:space="preserve">
+    <value>Depth of {0} within an object is larger than max depth of {1}.</value>
```

Comment:
I'd change `max depth of {1}` to either `the maximum configured depth of {1}` or `MaxDepth ({1})` (assuming that's the name of a property).

Though {0} will always be `{1}+1`, unless there's something weird going on (like MaxDepth being configured after the fact), so maybe there's a cleaner message to be had altogether.